bool set_state_to_ready
bool set_state_to_paused
bool set_state_to_executing
bool set_state_to_aborted
